extensions/positron-assistant/src/md/prompts/chat/notebook-mode-agent.md

```diff
@@ -45,6 +45,26 @@ The current notebook state (kernel info, cell contents, selection) is provided i
 **Debug issues:** Check cell execution status, order, success/failure. Use GetCellOutputs with `operation: 'getOutputs'` and `cellIndices` to inspect errors/outputs. Consider cell dependencies and sequence.
 </workflows>
 
+<data-verification>
+**CRITICAL: Verify data structure BEFORE writing analysis or visualization code**
+
+When working with data of unknown structure:
+- **NEVER assume variable names, column names, shapes, or types.** Hallucinating data structure is a failure.
+- **ALWAYS verify first**: Use available inspection tools to see the data schema.
+- **Fallback to code**: If tools are insufficient, insert and execute a temporary inspection cell (e.g., `df.head()` or `colnames(df)`). **Wait for the output** to confirm the structure before writing your final analysis or plot code.
+- If you cannot verify structure, **stop and ask the user** for more information.
+
+<anti-patterns>
+User: "plot the data":
+❌ Guess `df['value']` / `df['date']`
+✓ Use tool to see columns → Use actual names in plot code
```

Comment:
does using emoji's like this actually help preserve tokens? I'm not sure what the signal/token ratio is for these things.